.github/workflows/push.yml

```diff
@@ -310,3 +310,37 @@ jobs:
             ${{ github.sha }} \
             $(git merge-base ${{ github.event.pull_request.base.sha }} ${{ github.sha }}) \
             > $GITHUB_STEP_SUMMARY
+  WASM32_WASI:
+    name: "WASM32_WASI_${{ matrix.debug && 'DEBUG' || 'RELEASE' }}_${{ matrix.zts && 'ZTS' || 'NTS' }}"
+    runs-on: ubuntu-20.04
+    env:
+      CC: clang
+      CXX: clang++
+      LD: wasm-ld
+      AR: llvm-ar
+      RANLIB: llvm-ranlib
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v3
+      - name: apt
+        uses: ./.github/actions/apt-wasi
+      - name: ./configure
+        uses: ./.github/actions/configure-wasi
+        with:
+          configurationParameters: >-
+            --${{ matrix.debug && 'enable' || 'disable' }}-debug
+            --${{ matrix.zts && 'enable' || 'disable' }}-zts
+      - name: make
+        env:
+          CFLAGS: "-D_WASI_EMULATED_GETPID -D_WASI_EMULATED_SIGNAL -D_WASI_EMULATED_PROCESS_CLOCKS"
+          LDFLAGS: "-lwasi-emulated-getpid -lwasi-emulated-signal -lwasi-emulated-process-clocks"
+        run: make -j$(/usr/bin/nproc) cgi cli
+      - name: Optimize
+        uses: ./.github/actions/optimize-wasi
+      # To be defined how to approach test coverage on this platform.
```

Comment:
All core tests (tests that does not require any unsupported extension) must pass or they must be exluded if they are expected to fail ATM.

> setjmp/longjmp emulation

Fibers are core part of the PHP language since PHP 8.0 release. Can the support be added from the day zero?

Reply:
> All core tests (tests that does not require any unsupported extension) must pass or they must be exluded if they are expected to fail ATM.

Will exclude tests that don't apply on this platform in a next pass.

> Fibers are core part of the PHP language since PHP 8.0 release. Can the support be added from the day zero?

We should look into that. If it's a prerequisite for this patch to land, then yes. If it's fine to iterate I'd prefer so, to not block the whole patch on smaller specifics, given the patch is already somewhat big.